hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/SysFileOverrideOp.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.spec.utilops;
+
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getFileContents;
+import static com.hedera.services.bdd.spec.utilops.CustomSpecAssert.allRunFor;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.updateLargeFile;
+import static com.hedera.services.bdd.suites.HapiSuite.GENESIS;
+import static com.hedera.services.bdd.suites.HapiSuite.ONE_HBAR;
+import static com.hedera.services.bdd.suites.utils.sysfiles.serdes.StandardSerdes.SYS_FILE_SERDES;
+import static java.util.Objects.requireNonNull;
+
+import com.google.protobuf.ByteString;
+import com.hedera.services.bdd.spec.HapiSpec;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.OptionalLong;
+import java.util.function.Supplier;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A utility operation to override the contents of a dynamic system file, remembering the initial contents for
+ * later restoration.
+ */
+public class SysFileOverrideOp extends UtilOp {
+    private static final Logger log = LogManager.getLogger(SysFileOverrideOp.class);
+
+    public enum Target {
+        FEES {
+            @Override
+            public long number() {
+                return 111L;
+            }
+        },
+        EXCHANGE_RATES {
+            @Override
+            public long number() {
+                return 112L;
+            }
+        },
+        THROTTLES {
+            @Override
+            public long number() {
+                return 123L;
+            }
+        },
+        PERMISSIONS {
+            @Override
+            public long number() {
+                return 122L;
+            }
+        };
+
+        public abstract long number();
+    }
+
+    private final Target target;
+    private final Supplier<String> overrideSupplier;
+
+    private boolean autoRestoring = true;
+    private byte[] originalContents;
+
+    /**
```

Comment:
Just out of curiosity, when would you use this?

Reply:
It's used [here](https://github.com/hashgraph/hedera-services/blob/14836-simple-throttle-overrides/hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/SteadyStateThrottlingTest.java#L103) to use the snapshot/restore functionality in this op; but to run several `@HapiTest` between snapshotting and restoring.